airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/json_schema_helper.py

```diff
@@ -45,20 +45,27 @@ def get_property(self, path: List[str]):
             node = node["properties"][segment]
         return node
 
-    def get_type_for_key_path(self, path: List[str]):
+    def get_type_for_key_path(self, path: List[str]) -> Set[str]:
+        type_ = []
         try:
-            return self.get_property(path)["type"]
+            field = self.get_property(path)
+            type_ = field.get("format", field["type"])
+            if not isinstance(type_, List):
+                type_ = [type_]
         except KeyError:
-            return None
+            pass
+        return set(type_)
 
     def get_cursor_value(self, record, cursor_path):
         type_ = self.get_type_for_key_path(path=cursor_path)
         value = reduce(lambda data, key: data[key], cursor_path, record)
         return self.parse_value(value, type_)
 
     @staticmethod
-    def parse_value(value, type_):
-        if type_ in ("datetime", "date-time"):
+    def parse_value(value: Any, type_: Set[str]):
```

Comment:
are these parameters well named? it seems like `value` is actually JSON Schema `type` and `type_` is `format`?

Reply:
type is something in between I would say, but ok, lets make `format`
value is an actual value, it can be 1, '1', 'sometext', 20.0, '2010-10-11'
format is what tells us how to parse it